agent/mapcall/src/main/java/randoop/instrument/ReplacementFileReader.java

```diff
@@ -0,0 +1,524 @@
+        throw new ReplacementFileException(msg);
+      }
+    }
+    String msg =
+        String.format("No package or class for replacement %s found on classpath", replacement);
+    throw new ReplacementFileException(msg);
+  }
+
+  /**
+   * Discovers and adds method replacements for the package replacement determined by the {@code
+   * original} and {@code replacement} strings found in the given {@code ClassLoader}.
+   *
+   * @param replacementMap the method replacement map to which new replacements are added
+   * @param original the original package name
+   * @param replacement the replacement package name
+   * @param jarFile the jar file to search
+   * @return true if a class in the replacement package is found in the jar, false otherwise
+   */
+  private static boolean addPackageReplacements(
+      ConcurrentHashMap<MethodDef, MethodDef> replacementMap,
+      String original,
+      String replacement,
+      JarFile jarFile)
+      throws ReplacementFileException {
+
+    boolean found = false;
+    Enumeration<JarEntry> entries = jarFile.entries();
+    while (entries.hasMoreElements()) {
+      JarEntry entry = entries.nextElement();
+      String filename = entry.getName();
+      if (filename.endsWith(".class") && filename.startsWith(replacement.replace('.', '/'))) {
+        final String classname =
+            filename.substring(replacement.length() + 1, filename.lastIndexOf(".class"));
+        final String originalClassname = original + "." + classname;
+        if (classExists(originalClassname)) {
+          try {
+            Class<?> replacementClass = Class.forName(replacement + "." + classname);
+            addClassReplacements(replacementMap, originalClassname, replacementClass);
+            found = true;
+          } catch (ClassNotFoundException e) {
+            // ignore
+          }
+        }
+      }
+    }
+    return found;
+  }
+
+  /**
+   * Adds method replacements for the classes in the replacement package to the replacement map.
+   *
+   * <p>Assumes that subpackage structure of original and replacement packages is identical.
+   *
+   * @param replacementMap the method replacement map
+   * @param original the name of the original package
+   * @param replacement the name of the replacement package
+   * @param replacementDirectory the directory for the replacement package, must be non-null
+   */
+  private static void addPackageReplacements(
+      ConcurrentHashMap<MethodDef, MethodDef> replacementMap,
+      String original,
+      String replacement,
+      File replacementDirectory)
+      throws ReplacementFileException {
+
+    for (String filename : replacementDirectory.list()) {
+      if (filename.endsWith(".class")) {
+        final String classname = filename.substring(0, filename.lastIndexOf(".class"));
+        final String originalClassname = original + "." + classname;
+        if (classExists(originalClassname)) {
```

Comment:
Why wouldn't it exist?  Throw an exception, or write a comment explaining how this can happen and why it is acceptable.